### DISCUSSION TITLE
`cellx` does not work if `map-cells` is used

### DISCUSSION THREAD

**Discussion by yoonghm** [PgBiel/typst-tablex#98] [Q&A] [answered]
I use the following code to render a table. I want to align the heading rows to the center and fill it with different colour. However, it does not k if `map-cells` is used.

```typst
#figure(
  tablex(
    columns: (1fr, 4fr),
    align: left,
    auto-vlines: false,
    header-rows: 1,
    map-cells: cell => {
      if cell.y == 0 {
        cell.align = right  // no effect
        cell.fill = blue    // no effect
        set text(size: 14pt, fill: red)
        strong(cell.content)
      }
      else {
        set text(size: 12pt)
        cell.content
      }
    },
  
    /* Header */
    // cellx(...) also has no effect
    cellx(fill: blue, align: center)[Index], cellx(fill: blue, align: center)[Example],
    [`A`  ], [Apple],
    [`B`  ], [Ball],
    [`C`  ], [Car],
  ),
  kind: table,
  caption: "English alphabet"
)
```

![image](https://github.com/PgBiel/typst-tablex/assets/4215991/1ada9aa7-2f78-4abc-a087-90a0f462f8e6)

**Comment by PgBiel** [comment, marked as answer]
Hello,

`map-cells` provides you a `cellx` as a parameter and expects you to return a `cellx`. When you return the cell's content, you are discarding all other properties from the `cellx`.

This means you need to do the following:

```
map-cells: cell => {
      if cell.y == 0 {
        cell.align = right  // no effect
        cell.fill = blue    // no effect
        cell.content = {
          set text(size: 14pt, fill: red)
          strong(cell.content)
        }
      }
      else {
        cell.content = {
          set text(size: 12pt)
          cell.content
        }
      }
      cell   // return the cell with the modified properties
    },
```

Note that there is no other way to provide data to tablex other than by returning the cell here - the `cell` variable is just a **copy** of the real cellx, so modifying the copy doesn't do anything _unless_ you return it. Hope that clears it up!

**Reply by yoonghm** [reply]
Thanks!
